main.go

```diff
@@ -1,50 +1,73 @@
 package main
 
-import "fmt"
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
 
-const RESTAURANT_NAME string = "Jo's coffee shop!"
+	"strings"
+)
 
-type Product struct {
-	ID    int
-	Name  string
-	Price int
-}
+const RESTAURANT_NAME string = "Jo's coffee shop!"
 
-func listProducts() []Product {
-	return []Product{
-		{
-			ID:    1,
-			Name:  "Latte",
-			Price: 150,
-		},
-		{
-			ID:    2,
-			Name:  "Filter coffee",
-			Price: 100,
-		},
-	}
-}
 func handleCommand(command string) {
-	switch command {
+	tokens := strings.Split(command, " ")
+	switch tokens[0] {
 	case "menu":
 		fmt.Println("Menu：")
 		lists := listProducts()
 		for _, product := range lists {
 			fmt.Printf("%d. %s, $%d\n", product.ID, product.Name, product.Price)
 		}
+	case "add":
+		if len(tokens) != 2 && len(tokens) != 3 {
+			fmt.Println("USAGE: add PRODUCT_ID [QUANTITY]")
+			return
+		}
+		productID, err := strconv.ParseInt(tokens[1], 10, 64)
+		if err != nil {
+			fmt.Println("USAGE: add PRODUCT_ID [QUANTITY]")
+			return
+		}
+		// 如果有兩個參數，才需要解析
+		var quantity int64 = 1
+		if len(tokens) == 3 {
+			quantity, err = strconv.ParseInt(tokens[2], 10, 64)
+			if err != nil {
+				fmt.Println("USAGE: add PRODUCT_ID [QUANTITY]")
+				return
+			}
+		}
```

Comment:
The error handling part is repeated three times. You could refactor this branch of the `switch` statement by implementing a helper function, e. g.:

```go
func parseAddCommand(tokens []string) (ok bool, productID int, quantity int)
```

Then you could only handle the error case once.